### PR TITLE
fix(ci): fix e2e compaction test script

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Build compaction test binary in the build phase to avoid building in the e2e test job

## Checklist

~- [ ] I have written necessary rustdoc comments~
~- [ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: ci/workflows/main-cron.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ auto-retry: &auto-retry
 
 steps:
   - label: "build"
-    command: "ci/scripts/build.sh -t ci-release -p ci-release"
+    command: "ci/scripts/build.sh -t ci-dev -p ci-dev"
```
Comment (BugenZhao) [inline ci/workflows/main-cron.yml:24]: 🤔👀 Why change this to `dev`?

Comment (StrikeW) [reply ci/workflows/main-cron.yml:24]: Previously it is dev. I changed it to release expecting the data ingestion phase will be short, but the building time becomes longer, and the total time is almost the same. So I change it back to dev

Comment (StrikeW) [reply ci/workflows/main-cron.yml]: Test again, release build is more promising.  https://buildkite.com/risingwavelabs/main-cron/builds/196

**Review by huangjw806 — APPROVED** [review]
LGTM

**Pull request merged**